sql-cli/sql_cli/project.py

```diff
@@ -80,51 +83,85 @@ def data_dir(self) -> Path:
+        config = Config(environment=DEFAULT_ENVIRONMENT, project_dir=self.directory)
+        global_config_filepath = config.get_global_config_filepath()
         config.write_value_to_yaml(
-            "general", "data_dir", self._data_dir.resolve().as_posix(), global_env_filepath
+            "general", "data_dir", self._data_dir.resolve().as_posix(), global_config_filepath
         )
+
         # If the `Airflow Home` directory does not exist, Airflow initialisation flow takes care of creating the
         # directory. We rely on this behaviour and hence do not raise an exception if the path specified as
         # `Airflow Home` does not exist.
+        if not Path.exists(self._airflow_dags_folder):
+            raise FileNotFoundError(f"Specified DAGs directory {self._airflow_dags_folder} does not exist.")
         config.write_value_to_yaml(
-            "airflow", "home", self._airflow_home.resolve().as_posix(), global_env_filepath
+            "airflow", "dags_folder", str(self._airflow_dags_folder.resolve()), global_config_filepath
         )
-        if not self._airflow_dags_folder.exists():
-            raise FileNotFoundError(
-                f"Specified DAGs directory {self._airflow_dags_folder.as_posix()} does not exist."
-            )
+
+    def _initialise_env_config(self, environment: str) -> None:
+        """
+        Initialises the environment YAML config file that includes environment-specific values.
+
+        :param environment: Environment to set the configuration
+        """
+        config = Config(environment=environment, project_dir=self.directory)
+        config_filepath = config.get_env_config_filepath()
+        # If the `Airflow Home` directory does not exist, Airflow initialisation flow takes care of creating the
+        # directory. We rely on this behaviour and hence do not raise an exception if the path specified as
+        # `Airflow Home` does not exist.
         config.write_value_to_yaml(
-            "airflow", "dags_folder", self._airflow_dags_folder.resolve().as_posix(), global_env_filepath
+            "airflow", "home", str(self.get_env_airflow_home(environment)), config_filepath
         )
 
-    def _remove_unnecessary_airflow_files(self) -> None:
+    @staticmethod
+    def _remove_unnecessary_airflow_files(airflow_home: Path) -> None:
         """
-        Delete Airflow generated paths which are not necessary for the SQL CLI (scheduler & webserver-related).
+        Delete Airflow generated paths which are not necessary for the desired Airflow home.
+
+        :param airflow_home: Path to Airflow home we want to clean up.
         """
-        logs_folder = self.airflow_home / "logs"
+        logs_folder = airflow_home / "logs"
         shutil.rmtree(logs_folder)
 
-        webserver_config = self.airflow_home / "webserver_config.py"
+        webserver_config = airflow_home / "webserver_config.py"
         webserver_config.unlink()
```

Comment:
This function should actually live in our airflow utils and maybe get called in `initialise_airflow` at the end. We get the airflow_home there as well as an arg. WDYT?